### PR TITLE
config: fix JSON unmarshaling for HostPort

### LINKED ISSUE #2129: JSON de-serialization error on /status API call
**What did you do?**
I called the `StatusAPI.Get()` of the Alertmanager's Go client in order to get a server status info.  https://github.com/prometheus/alertmanager/blob/master/client/client.go#L125

**What did you expect to see?**
No errors being returned.

**What did you see instead? Under which circumstances?**
An error being returned in the format of `json: cannot unmarshal string into Go struct field GlobalConfig.configJSON.global.smtp_smarthost of type config.HostPort`.

**Environment**

* System information:

	Centos 7

* Alertmanager version:

	0.20.0-rc0

* Alertmanager configuration file:
```
global:
  http_config: {}
  smtp_from: prometheus@localhost
  smtp_require_tls: false
  smtp_smarthost: localhost:25

route:
  receiver: "test"
  group_by: ['alertname']
  group_wait:      15s
  group_interval:  15s
  repeat_interval: 1m

receivers:
- name: "test"
  webhook_configs:
  - url: 'http://webhook.service:8080/api/webhooks/prometheus'
    send_resolved: true
    http_config:
      basic_auth:
        username: admin@localhost
        password: test
```

### PR DESCRIPTION
**Pull request by simonpasquier**
Closes #2129

### CODE REVIEW

File: client/client_test.go [hunk 1]
```diff
@@ -90,9 +90,18 @@ func TestAPI(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
 	statusData := &ServerStatus{
-		ConfigYAML:    "{}",
-		ConfigJSON:    &config.Config{},
+		ConfigYAML: "{}",
+		ConfigJSON: &config.Config{
+			Global: &config.GlobalConfig{
+				PagerdutyURL:  &config.URL{u},
```
Comment (stuartnelson3) [inline client/client_test.go]: `composites: `github.com/prometheus/alertmanager/config.URL` composite literal uses unkeyed fields (govet)`

Comment (stuartnelson3) [reply client/client_test.go]: I would guess this means `u` needs to be given a key, but .. never seen this error before so :)

Comment (simonpasquier) [reply client/client_test.go]: Exactly!

**Comment by simonpasquier** [comment]
@stuartnelson3 friendly ping

**Pull request merged**